pkg/artifactory/datasource_artifactory_file_test.go

```diff
@@ -13,46 +16,148 @@ import (
+*/
 func TestDlFile(t *testing.T) {
+	downloadPath := getTempDownloadPath(t)
+
 	// every instance of RT has this repo and file out-of-the-box
-	script := `
+	const script = `
 		data "artifactory_file" "example" {
 		  repository      = "example-repo-local"
 		  path            = "crash.zip"
-		  output_path     = "${path.cwd}/crash.zip"
+		  output_path     = "%s"
 		  force_overwrite = true
 		}
 	`
+
+	var downloadCheck = func(state *terraform.State) error {
+		download := state.Modules[0].Resources["data.artifactory_file.example"].Primary.Attributes["output_path"]
+		_, err := os.Stat(download)
+		if err != nil {
+			return err
+		}
+		verified, err := VerifySha256Checksum(download, "7a2489dd209d0acb72f7f11d171b418e65648b9cc96c6c351e00e22551fdd8f1")
+		if !verified {
+			return fmt.Errorf("%s checksum does not have expected checksum", download)
+		}
+		return err
+	}
+
 	resource.Test(t, resource.TestCase{
-		PreCheck: func() {
-			testAccPreCheck(t)
-			client := getTestResty(t)
-			err := uploadTestFile(client, "../../samples/crash.zip", "example-repo-local/crash.zip", "application/zip")
-			if err != nil {
-				panic(err)
-			}
+		PreCheck:          downloadPreCheck(t, downloadPath),
+		ProviderFactories: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(script, downloadPath),
+				Check:  downloadCheck,
+			},
 		},
+	})
+}
+
+/*
+Negative test case on file download skip
+When file is present at output_path, checksum of files at output_path & repository path matches
+artifactory_file datasource will skip the download.
+*/
+func TestFileDownloadSkipCheck(t *testing.T) {
+	downloadPath := getTempDownloadPath(t)
+
+	// every instance of RT has this repo and file out-of-the-box
+	const script = `
+		data "artifactory_file" "example" {
+		  repository      = "example-repo-local"
+		  path            = "crash.zip"
+		  output_path     = "%s"
+		  force_overwrite = false
+		}
+	`
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          downloadPreCheck(t, downloadPath),
 		ProviderFactories: testAccProviders,
 		Steps: []resource.TestStep{
 			{
-				Config: script,
-				Check: func(state *terraform.State) error {
-					download := state.Modules[0].Resources["data.artifactory_file.example"].Primary.Attributes["output_path"]
-					_, err := os.Stat(download)
-					if err != nil {
-						return err
-					}
-					verified, err := VerifySha256Checksum(download, "7a2489dd209d0acb72f7f11d171b418e65648b9cc96c6c351e00e22551fdd8f1")
-
-					if !verified {
-						return fmt.Errorf("%s checksum does not have expected checksum", download)
-					}
-					return err
-				},
+				Config:      fmt.Sprintf(script, downloadPath),
+				ExpectError: regexp.MustCompile("err001: file download skiped"),
 			},
 		},
 	})
 }
+
+//Creates new directory tree if not exist
+func createNewDir(srcPath string) error {
+	if _, err := os.Stat(srcPath); errors.Is(err, os.ErrNotExist) {
```

Comment:
And if it's a different error (like permission denied) then you'd return nil ?? If it's not the one you care about, then you return the error you got. So you need 2 nested `if`s

Reply:
Improvement done
```
//Creates new directory tree if not exist
func createNewDir(srcPath string) error {
	_, err := os.Stat(srcPath)
	if err != nil {
		if errors.Is(err, os.ErrNotExist) {
			errMkDirAll := os.MkdirAll(srcPath, os.ModePerm)
			if errMkDirAll != nil {
				return errMkDirAll
			}
		} else {
			return err
		}
	}
	return nil
}
```